### ISSUE TITLE
OSError: broken data stream when reading image file

### ISSUE THREAD

**Issue by geilmaker** [python-pillow/Pillow#5834]
### What did you do?
test.py inside app
```python
from flask import Flask
from PIL import Image

app = Flask(__name__)

@app.route('/')
def hello():
    i = Image.open("<mypath>/img_4279.jpg")
    i.save("<mypath>/1.png")
    return 'Hello, World!'

if __name__ == "__main__":
    # app.debug = True
    app.run(host="<myip>")
```

web.wsgi
```python
import sys
sys.path.insert(0,"/var/www/flask/diashow_app/")
from app.test import app as application
```

### What did you expect to happen?
The script should read that image and just copy a png from it (as example).

### What actually happened?
When running that script on my server with apache it fails with
OSError: broken data stream when reading image file
When running that script on my server with python (without wsgi) as development server it works just fine.
Writing that code in python also works fine.
Running that exact program on my virtual machine (also Ubuntu 18.04) works fine with apache, no OSError there. 

### What have you tried?
When adding ImageFile.LOAD_TRUNCATED_IMAGES the OSError doesn't appear anymore, but the images are black and useless for my use.
The Program works fine when reading png files, the error just appear when using jpg/jpeg.
I installed the wsgi mod via apt-get, that didnt worked, now I installed it via pip and referenced it inside apache/mods-available and enabled it, but no difference.
Installed apache2 and apache2-dev.
I also tried to read the image with
```python
img = cv2.imread("<path-to-img>")
i = Image.fromarray(img)
```
but that ended in `AttributeError: 'NoneType' object has no attribute '__array_interface__'` at Image.fromarray(img).... so there is no array read by cv2...
I have really no idea anymore what I should do, just looked over all google pages found without chinese chars...

### What are your OS, Python and Pillow versions?
* OS: Ubuntu 18.04
* Python: 3.6.9
* Pillow: 8.4.0

If I missed to give any necessary information I'm sorry, pls just request... I have no idea how to fix that.

app.conf
```
WSGIApplicationGroup %{GLOBAL}
WSGIDaemonProcess FlaskApp user=www-data group=www-data threads=5 home=/var/www/flask/diashow_app/ python-path=/var/www/flask/diashow_app:/var/www/flask/diashow_app/app/venv/lib/python3.6/site-packages
WSGIProcessGroup FlaskApp

<VirtualHost *:80>
  ServerName domain.com
  ServerAdmin webmaster@domain.com

  WSGIScriptAlias / /var/www/flask/diashow_app/web.wsgi

  DocumentRoot /var/www/flask/diashow_app/
  <Directory /var/www/flask/diashow_app/app/>
    Order allow,deny
    Allow from all
  </Directory>
  Alias /static /var/www/flask/diashow_app/app/static
  <Directory /var/www/flask/diashow_app/app/static/>
    Order allow,deny
    Allow from all
  </Directory>
  ErrorLog ${APACHE_LOG_DIR}/error.log
  LogLevel warn
  CustomLog ${APACHE_LOG_DIR}/access.log combined
</VirtualHost>

<VirtualHost *:443>
  ServerName domain.com
  ServerAdmin webmaster@domain.com

  WSGIScriptAlias / /var/www/flask/diashow_app/web.wsgi

  DocumentRoot /var/www/flask/diashow_app/
  <Directory /var/www/flask/diashow_app/app/>
    Order allow,deny
    Allow from all
  </Directory>
  Alias /static /var/www/flask/diashow_app/app/static
  <Directory /var/www/flask/diashow_app/app/static/>
    Order allow,deny
    Allow from all
  </Directory>
  ErrorLog ${APACHE_LOG_DIR}/error.log
  LogLevel warn
  CustomLog ${APACHE_LOG_DIR}/access.log combined
SSLCertificateFile /etc/letsencrypt/live/domain.com/fullchain.pem
SSLCertificateKeyFile /etc/letsencrypt/live/domain.com/privkey.pem
Include /etc/letsencrypt/options-ssl-apache.conf
</VirtualHost>
```

**Comment by radarhere** [comment]
To check my understanding - this setup works in a virtual machine, but when you try it on your intended server, with the exact same setup as far as you can tell, it fails?

Could we get a copy of img_4279.jpg?

Is <mypath> absolute or relative?

**Comment by geilmaker** [comment]
Hello, thanks for your reply :)

Yea indeed, so my virtual machine was brand new Ubuntu 18.04 and just apache and this web app and docker installed. The intended server also got nextcloud and limesurvey installed, but its exact same apache version, python version, pillow version, ...
I tried the absolute path "var/www/flask/diashow_app/app/uploads/IMG_4279.jpg' and os.getcwd() (returns "var/www/flask/diashow_app/" and used os.path.join(os.getcwd(), "app/uploads/IMG_4279.JPG"). I also changed the path one time and then there was an error finding the file, so I suggest the script successfully finds the file. png's can be loaded with same technique successfull :)

Of course, that one was an personal img, so I just downloaded one from google and tested it with that one, but its also not working, as expected.
![img_4279](https://user-images.githubusercontent.com/34423885/141702196-94231e45-a8e2-40b8-ae03-9075052b83d8.jpg)

**Comment by geilmaker** [comment]
And this script is working fine if started with python test.py, the development server is starting on port 5000 and when I connect on it everything works fine. There is really just this one error, that jpg's cant be processed when the script is started via apache with wsgi script.... But png's can, so its not a general problem.... you know what I mean? ...so this is really damaging my brain :D

**Comment by radarhere** [comment]
If you
```python
import sys
print(sys.version)
```
does that return different values based on how you run it?

**Comment by geilmaker** [comment]
Unfortunatly not. My VM and the Server are both printing 3.6.9 and [GCC 8.4.0]

**Comment by geilmaker** [comment]
```
[Mon Nov 15 00:40:48.201522 2021] [wsgi:error] [pid 28799] [remote 84.154.213.199:53312] 3.6.9 (default, Jan 26 2021, 15:33:00) 
[Mon Nov 15 00:40:48.201575 2021] [wsgi:error] [pid 28799] [remote 84.154.213.199:53312] [GCC 8.4.0]
[Mon Nov 15 00:40:48.201584 2021] [wsgi:error] [pid 28799] [remote 84.154.213.199:53312] 
[Mon Nov 15 00:40:48.201631 2021] [wsgi:error] [pid 28799] [remote 84.154.213.199:53312] True
[Mon Nov 15 00:40:48.235290 2021] [wsgi:error] [pid 28799] [remote 84.154.213.199:53312] [2021-11-15 00:40:48,233] ERROR in app: Exception on / [GET]
[Mon Nov 15 00:40:48.235339 2021] [wsgi:error] [pid 28799] [remote 84.154.213.199:53312] Traceback (most recent call last):
[Mon Nov 15 00:40:48.235346 2021] [wsgi:error] [pid 28799] [remote 84.154.213.199:53312]   File "/var/www/flask/diashow_app/app/venv/lib/python3.6/site-packages/flask/app.py", line 2073, in wsgi_app
[Mon Nov 15 00:40:48.235351 2021] [wsgi:error] [pid 28799] [remote 84.154.213.199:53312]     response = self.full_dispatch_request()
[Mon Nov 15 00:40:48.235357 2021] [wsgi:error] [pid 28799] [remote 84.154.213.199:53312]   File "/var/www/flask/diashow_app/app/venv/lib/python3.6/site-packages/flask/app.py", line 1518, in full_dispatch_request
[Mon Nov 15 00:40:48.235362 2021] [wsgi:error] [pid 28799] [remote 84.154.213.199:53312]     rv = self.handle_user_exception(e)
[Mon Nov 15 00:40:48.235368 2021] [wsgi:error] [pid 28799] [remote 84.154.213.199:53312]   File "/var/www/flask/diashow_app/app/venv/lib/python3.6/site-packages/flask/app.py", line 1516, in full_dispatch_request
[Mon Nov 15 00:40:48.235373 2021] [wsgi:error] [pid 28799] [remote 84.154.213.199:53312]     rv = self.dispatch_request()
[Mon Nov 15 00:40:48.235378 2021] [wsgi:error] [pid 28799] [remote 84.154.213.199:53312]   File "/var/www/flask/diashow_app/app/venv/lib/python3.6/site-packages/flask/app.py", line 1502, in dispatch_request
[Mon Nov 15 00:40:48.235383 2021] [wsgi:error] [pid 28799] [remote 84.154.213.199:53312]     return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
[Mon Nov 15 00:40:48.235388 2021] [wsgi:error] [pid 28799] [remote 84.154.213.199:53312]   File "/var/www/flask/diashow_app/app/test.py", line 20, in hello
[Mon Nov 15 00:40:48.235392 2021] [wsgi:error] [pid 28799] [remote 84.154.213.199:53312]     img = img.resize((1920, 1080), Image.ANTIALIAS)
[Mon Nov 15 00:40:48.235397 2021] [wsgi:error] [pid 28799] [remote 84.154.213.199:53312]   File "/var/www/flask/diashow_app/app/venv/lib/python3.6/site-packages/PIL/Image.py", line 1989, in resize
[Mon Nov 15 00:40:48.235402 2021] [wsgi:error] [pid 28799] [remote 84.154.213.199:53312]     self.load()
[Mon Nov 15 00:40:48.235406 2021] [wsgi:error] [pid 28799] [remote 84.154.213.199:53312]   File "/var/www/flask/diashow_app/app/venv/lib/python3.6/site-packages/PIL/ImageFile.py", line 274, in load
[Mon Nov 15 00:40:48.235411 2021] [wsgi:error] [pid 28799] [remote 84.154.213.199:53312]     raise_oserror(err_code)
[Mon Nov 15 00:40:48.235416 2021] [wsgi:error] [pid 28799] [remote 84.154.213.199:53312]   File "/var/www/flask/diashow_app/app/venv/lib/python3.6/site-packages/PIL/ImageFile.py", line 67, in raise_oserror
[Mon Nov 15 00:40:48.235421 2021] [wsgi:error] [pid 28799] [remote 84.154.213.199:53312]     raise OSError(message + " when reading image file")
[Mon Nov 15 00:40:48.235427 2021] [wsgi:error] [pid 28799] [remote 84.154.213.199:53312] OSError: broken data stream when reading image file
```
And this is the error.log of apache if you are interested. In comparison to my code written at the top I changed the save() with resize(), it was just a try, but all those methods are calling the load() method inside, which is leading to the error.

**Comment by geilmaker** [comment]
I found one answer between so many others not working in another issue, but I couldnt test this one because I dont know how. I dont know what to change to see if that answer is working for me.
https://github.com/python-pillow/Pillow/issues/1510#issuecomment-152697768

**Comment by radarhere** [comment]
#1510 is talking about a specific image that won't load in Pillow at all.
You're saying that all JPEG images won't load for your specific environment.
I don't see much of a connection.

Is the Pillow version the same in both? Are the JPEG versions the same?
```python
from PIL import features
features.pilinfo(None, False)
```

**Comment by geilmaker** [comment]
Looks like they are pretty much the same, also the jpeg support is ok and they have the same version (compiled for 9.0)

Thats the log on my server
```
[Mon Nov 15 03:41:26.517594 2021] [wsgi:error] [pid 29837] --------------------------------------------------------------------
[Mon Nov 15 03:41:26.517647 2021] [wsgi:error] [pid 29837] Pillow 8.4.0
[Mon Nov 15 03:41:26.517664 2021] [wsgi:error] [pid 29837] Python 3.6.9 (default, Jan 26 2021, 15:33:00)
[Mon Nov 15 03:41:26.517675 2021] [wsgi:error] [pid 29837]        [GCC 8.4.0]
[Mon Nov 15 03:41:26.517684 2021] [wsgi:error] [pid 29837] --------------------------------------------------------------------
[Mon Nov 15 03:41:26.517717 2021] [wsgi:error] [pid 29837] Python modules loaded from /var/www/flask/diashow_app/app/venv/lib/python3.6/site-packages/PIL
[Mon Nov 15 03:41:26.517732 2021] [wsgi:error] [pid 29837] Binary modules loaded from /var/www/flask/diashow_app/app/venv/lib/python3.6/site-packages/PIL
[Mon Nov 15 03:41:26.517747 2021] [wsgi:error] [pid 29837] --------------------------------------------------------------------
[Mon Nov 15 03:41:26.517785 2021] [wsgi:error] [pid 29837] --- PIL CORE support ok, compiled for 8.4.0
[Mon Nov 15 03:41:26.593761 2021] [wsgi:error] [pid 29837] --- TKINTER support ok, loaded 8.6
[Mon Nov 15 03:41:26.627997 2021] [wsgi:error] [pid 29837] --- FREETYPE2 support ok, loaded 2.8.1
[Mon Nov 15 03:41:26.638216 2021] [wsgi:error] [pid 29837] --- LITTLECMS2 support ok, loaded 2.12
[Mon Nov 15 03:41:26.638273 2021] [wsgi:error] [pid 29837] --- WEBP support ok, loaded 0.6.1
[Mon Nov 15 03:41:26.638298 2021] [wsgi:error] [pid 29837] --- WEBP Transparency support ok
[Mon Nov 15 03:41:26.638316 2021] [wsgi:error] [pid 29837] --- WEBPMUX support ok
[Mon Nov 15 03:41:26.638333 2021] [wsgi:error] [pid 29837] --- WEBP Animation support ok
[Mon Nov 15 03:41:26.638397 2021] [wsgi:error] [pid 29837] --- JPEG support ok, compiled for 9.0
[Mon Nov 15 03:41:26.638437 2021] [wsgi:error] [pid 29837] --- OPENJPEG (JPEG2000) support ok, loaded 2.4.0
[Mon Nov 15 03:41:26.638477 2021] [wsgi:error] [pid 29837] --- ZLIB (PNG/ZIP) support ok, loaded 1.2.11
[Mon Nov 15 03:41:26.638518 2021] [wsgi:error] [pid 29837] --- LIBTIFF support ok, loaded 4.0.9
[Mon Nov 15 03:41:26.638558 2021] [wsgi:error] [pid 29837] --- RAQM (Bidirectional Text) support ok, loaded 0.7.2, fribidi 0.19.7, harfbuzz 3.0.0
[Mon Nov 15 03:41:26.638576 2021] [wsgi:error] [pid 29837] *** LIBIMAGEQUANT (Quantization method) support not installed
[Mon Nov 15 03:41:26.638592 2021] [wsgi:error] [pid 29837] --- XCB (X protocol) support ok
[Mon Nov 15 03:41:26.638602 2021] [wsgi:error] [pid 29837] --------------------------------------------------------------------
```

The following on my VM
```
Mon Nov 15 02:46:02.630934 2021] [wsgi:error] [pid 2303:tid 140262517393152] --------------------------------------------------------------------
[Mon Nov 15 02:46:02.630948 2021] [wsgi:error] [pid 2303:tid 140262517393152] Pillow 8.4.0
[Mon Nov 15 02:46:02.630955 2021] [wsgi:error] [pid 2303:tid 140262517393152] Python 3.6.9 (default, Jan 26 2021, 15:33:00)
[Mon Nov 15 02:46:02.630959 2021] [wsgi:error] [pid 2303:tid 140262517393152]        [GCC 8.4.0]
[Mon Nov 15 02:46:02.630963 2021] [wsgi:error] [pid 2303:tid 140262517393152] --------------------------------------------------------------------
[Mon Nov 15 02:46:02.630976 2021] [wsgi:error] [pid 2303:tid 140262517393152] Python modules loaded from /var/www/flask/diashow_app/app/venv/lib/python3.6/site-packages/PIL
[Mon Nov 15 02:46:02.630983 2021] [wsgi:error] [pid 2303:tid 140262517393152] Binary modules loaded from /var/www/flask/diashow_app/app/venv/lib/python3.6/site-packages/PIL
[Mon Nov 15 02:46:02.630986 2021] [wsgi:error] [pid 2303:tid 140262517393152] --------------------------------------------------------------------
[Mon Nov 15 02:46:02.631007 2021] [wsgi:error] [pid 2303:tid 140262517393152] --- PIL CORE support ok, compiled for 8.4.0
[Mon Nov 15 02:46:02.631742 2021] [wsgi:error] [pid 2303:tid 140262517393152] *** TKINTER support not installed
[Mon Nov 15 02:46:02.646339 2021] [wsgi:error] [pid 2303:tid 140262517393152] --- FREETYPE2 support ok, loaded 2.11.0
[Mon Nov 15 02:46:02.649088 2021] [wsgi:error] [pid 2303:tid 140262517393152] --- LITTLECMS2 support ok, loaded 2.12
[Mon Nov 15 02:46:02.649104 2021] [wsgi:error] [pid 2303:tid 140262517393152] --- WEBP support ok, loaded 1.2.1
[Mon Nov 15 02:46:02.649118 2021] [wsgi:error] [pid 2303:tid 140262517393152] --- WEBP Transparency support ok
[Mon Nov 15 02:46:02.649143 2021] [wsgi:error] [pid 2303:tid 140262517393152] --- WEBPMUX support ok
[Mon Nov 15 02:46:02.649155 2021] [wsgi:error] [pid 2303:tid 140262517393152] --- WEBP Animation support ok
[Mon Nov 15 02:46:02.649188 2021] [wsgi:error] [pid 2303:tid 140262517393152] --- JPEG support ok, compiled for 9.0
[Mon Nov 15 02:46:02.649210 2021] [wsgi:error] [pid 2303:tid 140262517393152] --- OPENJPEG (JPEG2000) support ok, loaded 2.4.0
[Mon Nov 15 02:46:02.649230 2021] [wsgi:error] [pid 2303:tid 140262517393152] --- ZLIB (PNG/ZIP) support ok, loaded 1.2.11
[Mon Nov 15 02:46:02.649251 2021] [wsgi:error] [pid 2303:tid 140262517393152] --- LIBTIFF support ok, loaded 4.3.0
[Mon Nov 15 02:46:02.649270 2021] [wsgi:error] [pid 2303:tid 140262517393152] --- RAQM (Bidirectional Text) support ok, loaded 0.7.2, fribidi 0.19.7, harfbuzz 3.0.0
[Mon Nov 15 02:46:02.649278 2021] [wsgi:error] [pid 2303:tid 140262517393152] *** LIBIMAGEQUANT (Quantization method) support not installed
[Mon Nov 15 02:46:02.649287 2021] [wsgi:error] [pid 2303:tid 140262517393152] --- XCB (X protocol) support ok
[Mon Nov 15 02:46:02.649291 2021] [wsgi:error] [pid 2303:tid 140262517393152] --------------------------------------------------------------------
```

**Comment by radarhere** [comment]
To suggest a crazy idea - if you try rename the JPEG file's extension to be '.png', does it work then?

**Comment by geilmaker** [comment]
Thats a great idea! :)

The print is via python and apache2 the same:
"File size: 241205"
My Windows also says Size: 241.205 Bytes and Size on Disk: 241.664 Bytes
So the 241205 match....

Started with apache2 the same error appears at the resize()/load() function, started with python it works again just fine :)

**Comment by radarhere** [comment]
So it's the same Pillow version running the same libjpeg version with the same input data... both on Linux... for a bug that I can't reproduce locally and you can't reproduce in a VM... you're not leaving me much to work with here :p

I'm guessing that Pillow was installed through pip? If you uninstall Pillow and then try and build it from source instead  - `python3 -m pip install Pillow --no-binary :all:` - does that fix the problem? (You may find that you have to install some [dependencies](https://pillow.readthedocs.io/en/stable/installation.html#building-on-linux) to get it working)

**Comment by geilmaker** [comment]
Ya im sorry for that, I dont really know whats happening too.... Do you know any apt packages that definetly need to be install, or some that shouldnt be? ...I guess I installed every possible package, but maybe there is an error there....

At the moment im not really sure if the error is rly inside Pillow, cuz why is everything fine when I start the exact script via python as production server? ....so I guess it needs to be something with apache and jpeg... but I got no idea left what should work....

Ya I tried that, there were no dependencies displayed that I should install, Pillow installed successfull but nothing changed :D

**Comment by geilmaker** [comment]
Okay, maybe - to be completely exactly - my VM is running
Ubuntu 18.04.6 LTS (GNU/Linux 4.15.0-162-generic x86_64)
and the Server is running
Ubuntu 18.04.6 LTS (GNU/Linux 4.15.0 x86_64)
Didnt realized that, doesnt look like an important difference for me, but I dont know what you guess :)

**Comment by geilmaker** [comment]
Left side the server, right side the vm...... So python packages are exactly the same....
![pipinstalled](https://user-images.githubusercontent.com/34423885/141882957-2f6cc9e9-034d-4bce-9f0c-8c35ce5c8471.png)

**Comment by nulano** [comment]
>To check that Pillow is getting the full file, if you try and read the file into memory first, is the file size what it should be?

>Thats a great idea! :)
>The print is via python and apache2 the same

I don't really expect to see a difference since the file length is the same, but if you compute the hash of the file, do you get the same result? From the screenshot it looks like you are using the same precompiled wheel, so I don't see how you could get different behaviour if it is the same data...

For example: https://stackoverflow.com/a/22058673/1648883 (simplified)

```python
import sys
import hashlib

sha1 = hashlib.sha1()

with open('the_file.jpg', 'rb') as f:
    while True:
        data = f.read(65536)
        if not data:
            break
        sha1.update(data)

print("SHA1: {0}".format(sha1.hexdigest()))
```

---

> Ya I tried that, there were no dependencies displayed that I should install, Pillow installed successfull but nothing changed :D

I think you have to uninstall Pillow first with `python3 -m pip uninstall Pillow` before running `python3 -m pip install Pillow --no-binary :all:` again.

**Comment by geilmaker** [comment]
Hello nulano, thanks for joining! :)

ApacheOnServer: SHA1: 5c27dfaafd85f2069b9263f5833d0fa5b9d53011
VM:                     SHA1: 5c27dfaafd85f2069b9263f5833d0fa5b9d53011
PythonOnServer: SHA1: 5c27dfaafd85f2069b9263f5833d0fa5b9d53011
So they are all 3 the same :|

> I think you have to uninstall Pillow first with python3 -m pip uninstall Pillow before running python3 -m pip install Pillow --no-binary :all: again.
Ya I did that, the following is my input line + the output from the shell
```
python3 -m pip install Pillow --no-cache-dir --no-binary :all:
Collecting Pillow
  Downloading Pillow-8.4.0.tar.gz (49.4 MB)
     |████████████████████████████████| 49.4 MB 3.6 MB/s
  Preparing metadata (setup.py) ... done
Skipping wheel build for Pillow, due to binaries being disabled for it.
Installing collected packages: Pillow
    Running setup.py install for Pillow ... done
Successfully installed Pillow-8.4.0
```

**Comment by geilmaker** [comment]
So maybe I should consider explaining what I really want to do with all of that, maybe you guys got another idea how to do it....
I just wanted a little python script that can create a diashow with uploaded images and audio, for that I was reading the files with Pillow, then resized them in the right ratio for 1920x1080 and pasted them with the right offset on a black Image with the exact size of 1920x1080, so I can create a video in opencv with 1920x1080...
But rly, cant it just work on the **** server with apache, when its working when started with python?!? Makes completely no sense to me....

**Comment by radarhere** [comment]
If you're planning to use OpenCV later, you might find it simpler to just use OpenCV to start with as well. I believe you can open an image, resize and paste using OpenCV - see https://stackoverflow.com/questions/14063070/overlay-a-smaller-image-on-a-larger-image-python-opencv for example.

**Comment by geilmaker** [comment]
I already have everything working with opencv and Pillow, there was just this error occuring, so I made a simple script with just that Pillow part and even that was failing, so I just uploaded the simple script :)
Thats a good advice, I tried to go for opencv at the beginning, but then I found Pillow and pasting was much easier, so I used this... I will try ur link when I get back from university, thanks for the advise!
CU later :D

**Comment by geilmaker** [comment]
So I tried doing it with cv2, but it also doesnt work, but now I know the problem is not inside Pillow, because its not used at the time. I broke my code down again to the following miniscript:
```python
from flask import Flask
import cv2 as cv

app = Flask(__name__)

@app.route('/')
def hello():
    img = cv.imread("/var/www/flask/diashow_app/app/uploads/99e3a388-a2de-4b52-b8f3-e824b1b72e27/img/pineapple_PNG2755.$    print(img.shape)
    img = cv.imread("/var/www/flask/diashow_app/app/uploads/99e3a388-a2de-4b52-b8f3-e824b1b72e27/img/img_4279.jpg")
    print(img.shape)
    return 'Hello, World!'

if __name__ == "__main__":
    # app.debug = True
    app.run(host = "<my-ip>")
```
When I run that script with python test.py everthing works fine, its printing me both image shapes.
When I run that script with the wsgi script via apache, the png is printing its shape, but at the print of the jpg shape its crashing because it cant use shape on None.... so the img returned by that cv.imread() is None, but just at the jpg file. And its working when I start the script with python...

My wsgi script to start the test.py inside the app folder:
```
#!/usr/bin/python3
import sys
sys.path.insert(0,"/var/www/flask/diashow_app/")
from app.test import app as application
```

Anybody an idea? ^^

**Comment by geilmaker** [comment]
I just found the error inside apache.
https://www.digitalocean.com/community/tutorials/how-to-configure-apache-http-with-mpm-event-and-php-fpm-on-ubuntu-18-04
On my server was mpm prefork activated, I just switched that to mpm event with that tutorial, now this diashow application is working fine, but nextcloud and limesurvey arent happy about those changes.... So thats kinda a problem...
With that in mind, does anybody know a possibility to run it in mpm prefork?

**Comment by radarhere** [comment]
Congratulations on figuring that out!
So do you have any idea why prefork is failing for you? Is it timing out? Are multiple processes trying to read the JPEG file at once?

**Comment by geilmaker** [comment]
So hello everybody!
I am a happy person, like I was 5 days ago :)
I found the issue inside apache mpm prefork, I dont know why but its not working with that one so I needed to switch to mpm event and now my script is working!
If somebody needs it, I used the following side to switch to mpm event:
https://www.digitalocean.com/community/tutorials/how-to-configure-apache-http-with-mpm-event-and-php-fpm-on-ubuntu-18-04
but that crashed my limesurvey and nextcloud instance, so I googled a bit and found the following solution:
https://help.nextcloud.com/t/help-with-php-fpm-on-apache/19544/7
Doin the exact same 4 lines described by MichaIng with the php7.4-fpm version worked perfectly for me. Now nextcloud is running without any noticing issues so far, limesurvey instance is up and my sweet application is working.... Finally!
Thanks guys for your help and advise, sorry for that issue that has nothing to do with pillow in the end...
Have a good one, cya :)
